src/ansi-c/expr2c.cpp

```diff
@@ -860,6 +840,42 @@ std::string expr2ct::convert_array_type(
     src.subtype(), qualifiers, declarator_str+array_suffix);
 }
 
+std::string expr2ct::convert_pointer_type(
+  const typet &src,
+  const c_qualifierst &qualifiers,
+  const std::string &declarator)
+{
+    c_qualifierst new_qualifiers(qualifiers);
+    new_qualifiers.read(src);
+    std::string q=new_qualifiers.as_string();
+
+    c_qualifierst sub_qualifiers;
+    sub_qualifiers.read(src.subtype());
+    const typet &subtype_followed=ns.follow(src.subtype());
+
+    // The star gets attached to the declarator.
+    std::string new_declarator="*";
+
+    if(q!="" &&
+       (!declarator.empty() || subtype_followed.id()==ID_pointer))
+      new_declarator+=" "+q;
+
+    new_declarator+=declarator;
+
+    // Depending on precedences, we may add parentheses.
+    if(subtype_followed.id()==ID_code ||
+        (sizeof_nesting==0 &&
+         (subtype_followed.id()==ID_array ||
+          subtype_followed.id()==ID_incomplete_array)))
+      new_declarator="("+new_declarator+")";
+
+    if(subtype_followed.id()==ID_array)
```

Comment:
This behaviour should be explained and also toggable to ensure that the behaviour of `expr2ct` is not changed. In this case, I think a Boolean toggle passed in is sufficient.